pkg/xds/generator/proxy_template.go

```diff
@@ -151,30 +153,54 @@ func (_ OutboundProxyGenerator) Generate(ctx xds_context.Context, proxy *model.P
 	resources := make([]*Resource, 0, len(ofaces))
 	names := make(map[string]bool)
 	sourceService := proxy.Dataplane.Spec.GetIdentifyingService()
-	for _, oface := range ofaces {
+	for i, oface := range ofaces {
 		endpoint, err := kuma_mesh.ParseOutboundInterface(oface.Interface)
 		if err != nil {
 			return nil, err
 		}
 
-		serviceTag := kuma_mesh.ServiceTagValue(oface.Service)
-		edsClusterName := outboundClusterName(serviceTag, oface.ServicePort)
-		if used := names[edsClusterName]; !used {
-			resources = append(resources, &Resource{
-				Name:     edsClusterName,
-				Resource: envoy.CreateEdsCluster(ctx, edsClusterName, proxy.Metadata),
-			})
-			resources = append(resources, &Resource{
-				Name:     edsClusterName,
-				Resource: envoy.CreateClusterLoadAssignment(edsClusterName, proxy.OutboundTargets[oface.Service]),
+		route := proxy.TrafficRoutes[oface.Service]
+		if route == nil {
+			return nil, errors.Errorf("outbound interface [%d]{service=%q} has no TrafficRoute", i, oface.Service)
+		}
+
+		var clusters []envoy.ClusterInfo
+		for _, destination := range route.Spec.Conf {
+			service, ok := destination.Destination[kuma_mesh.ServiceTag]
+			if !ok {
+				// all destinations must have `service` tag
```

Comment:
can we then return an error since this should not happen?

Reply:
done